pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1332 @@
+				default:
+					// If they worked okay, we're building, we can update our status?
+					glog.Infof("A build %s is in progress (%s) for pool %s", mostRecentBuild.Name, mostRecentBuild.Status.Phase, pool.Name)
+				}
+
+			}
+		}
+
+	}
+
+	// Do we have an image stream for this pool? We should if we got here.
+	is, err := ctrl.isLister.ImageStreams(ctrlcommon.MCONamespace).Get(poolImageStream.Name)
+	if apierrors.IsNotFound(err) {
+		// TODO(jkyros): As cgwalters points out, this should probably degrade because it should exist
+		glog.Warningf("ImageStream for %s does not exist (yet?): %s", pool.Name, err)
+	} else {
+		// If there is an image ready, annotate the pool with it so node controller can use it if it's the right one
+		err := ctrl.annotatePoolWithNewestImage(is, pool)
+		if err != nil {
+			return err
+		}
+	}
+
+	// TODO(jkyros): Only update if we changed, don't always update. Also, if we update here and then update status again, that seems
+	// wasteful.
+	_, err = ctrl.client.MachineconfigurationV1().MachineConfigPools().Update(context.TODO(), pool, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.syncAvailableStatus(pool)
+
+}
+
+// Machine Config Pools
+
+func (ctrl *Controller) addMachineConfigPool(obj interface{}) {
+	pool := obj.(*mcfgv1.MachineConfigPool)
+	glog.V(4).Infof("Adding MachineConfigPool %s", pool.Name)
+	ctrl.enqueueMachineConfigPool(pool)
+
+}
+
+func (ctrl *Controller) updateMachineConfigPool(old, cur interface{}) {
+	oldPool := old.(*mcfgv1.MachineConfigPool)
+	curPool := cur.(*mcfgv1.MachineConfigPool)
+
+	glog.V(4).Infof("Updating MachineConfigPool %s", oldPool.Name)
+	ctrl.enqueueMachineConfigPool(curPool)
+}
+
+func (ctrl *Controller) deleteMachineConfigPool(obj interface{}) {
+	pool, ok := obj.(*mcfgv1.MachineConfigPool)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		pool, ok = tombstone.Obj.(*mcfgv1.MachineConfigPool)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a MachineConfigPool %#v", obj))
+			return
+		}
+	}
+	glog.V(4).Infof("Deleting MachineConfigPool %s", pool.Name)
+}
+
+// ImagStreams
+
+func (ctrl *Controller) addImageStream(obj interface{}) {
+
+}
+
+func (ctrl *Controller) updateImageStream(old, cur interface{}) {
+	imagestream := cur.(*imagev1.ImageStream)
+	controllerRef := metav1.GetControllerOf(imagestream)
+
+	if controllerRef != nil {
+
+		if pool := ctrl.resolveControllerRef(controllerRef); pool != nil {
+
+			glog.Infof("ImageStream %s changed for pool %s", imagestream.Name, pool.Name)
+
+			// TODO(jkyros): This is a race I usually win, but I won't always, and we need a better
+			// way to get this metadata in
+			if imagestream.Name == pool.Name+ctrlcommon.ImageStreamSuffixRenderedConfig {
+				ctrl.cheatMachineConfigLabelIntoBuildConfig(imagestream, pool)
```

Comment:
Yes, this is silly. This is me trying to sneak the dynamic labels onto the buildconfig before it triggers. 

I'm injecting the image labels via the dockerfile + args right now, but if a user runs `oc start-build`, say to trigger after a build fails, the user's build won't have the args, which means it won't get the label, which means we won't know what machineconfig an image contains (and I know we don't *need* it anymore because our config is the image, but I really do want to be able to show it back somehow).